### PR TITLE
[2기-C] 정하영 - PostRepository 테스트와 구현

### PR DESCRIPTION
**Pull request by gkdud583**
* PostRepository 테스트와 구현입니다.
* @Sql로 직접 작성한 쿼리를 확인해 보기 위해 테스트를 추가하였습니다.
* Post 조회시 User 정보도 함께 조회해 오기 위해 페치 조인을 사용하였습니다.
* 아래는 콘솔로 확인해 본 실행된 sql입니다.

 **findWithPagination**
select post0_.id as id1_0_0_, user1_.id as id1_1_1_, post0_.created_at as created_2_0_0_, post0_.modified_at as modified3_0_0_, post0_.content as content4_0_0_, post0_.title as title5_0_0_, post0_.user_id as user_id6_0_0_, user1_.created_at as created_2_1_1_, user1_.modified_at as modified3_1_1_, user1_.age as age4_1_1_, user1_.hobby as hobby5_1_1_, user1_.name as name6_1_1_ from post post0_ inner join user user1_ on post0_.user_id=user1_.id limit ?

**findById**
select post0_.id as id1_0_0_, user1_.id as id1_1_1_, post0_.created_at as created_2_0_0_, post0_.modified_at as modified3_0_0_, post0_.content as content4_0_0_, post0_.title as title5_0_0_, post0_.user_id as user_id6_0_0_, user1_.created_at as created_2_1_1_, user1_.modified_at as modified3_1_1_, user1_.age as age4_1_1_, user1_.hobby as hobby5_1_1_, user1_.name as name6_1_1_ from post post0_ inner join user user1_ on post0_.user_id=user1_.id where post0_.id=?

### CODE REVIEW

File: src/main/java/com/example/jpaboard/domain/post/PostRepository.java [hunk 1]
```diff
@@ -2,12 +2,15 @@
...
 import java.util.List;
 import java.util.Optional;
 
 public interface PostRepository extends JpaRepository<Post, Long> {
+    @Query("select p from Post p join fetch p.user")
     List<Post> findWithPagination(Pageable pageable);
 
+    @Query("select p from Post p join fetch p.user where p.id = :id")
     Optional<Post> findById(@Param("id") Long id);
 }
```
Comment (waterfogSW) [inline src/main/java/com/example/jpaboard/domain/post/PostRepository.java:16]: join fetch! 참고하겠습니다 👍🏼

**Review by waterfogSW — APPROVED** [review]
잘했어요박수잘했어김정은정은이축하해하하하아이좋아GIF

**Review by youngjunryu — APPROVED** [review]

File: src/test/java/com/example/jpaboard/domain/post/PostRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            postRepository.saveAllAndFlush(Arrays.asList(new Post("제목1", "내용1", user), new Post("제목1", "내용1", user)));
+            PageRequest pageRequest = PageRequest.of(0, 10);
+
+            //when
+            List<Post> posts = postRepository.findWithPagination(pageRequest);
+
+            //then
+            Assertions.assertThat(posts.size()).isEqualTo(2);
+            Assertions.assertThat(em.contains(posts.get(0).getUser())).isTrue();
+            Assertions.assertThat(em.contains(posts.get(1).getUser())).isTrue();
```
Comment (UJ15) [inline src/test/java/com/example/jpaboard/domain/post/PostRepositoryTest.java:43]: 엔티티 매니저를 활용해서 테스트를 할 수 있군요 👍

**Review by UJ15 — APPROVED** [review]
6698a827595bf05163d68a1e85b05898

**Pull request merged**